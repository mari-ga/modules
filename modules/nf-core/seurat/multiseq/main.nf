process SEURAT_MULTISEQ {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/r-seurat:3.0.2--r36h0357c0b_1':
        'quay.io/biocontainers/r-seurat:3.0.2--r36h0357c0b_0' }"

    input:
    tuple val(meta), path(hto_matrix)
    tuple val(meta), path(rna_matrix)
    val produce_plots
    output:
    tuple val(meta), path("*.csv")                  ,emit: assignment
    path "versions.yml"                             ,emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    template 'MULTIseq.R'
}
