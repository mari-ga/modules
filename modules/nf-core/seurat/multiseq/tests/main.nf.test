// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test seurat/multiseq
nextflow_process {

    name "Test Process SEURAT_MULTISEQ"
    script "../main.nf"
    process "SEURAT_MULTISEQ"

    tag "modules"
    tag "modules_nfcore"
    tag "seurat"
    tag "seurat/multiseq"
    tag "untar"

    test("Standard_Multiome - 10x mtx") {
        config "./nextflow.config"
        setup {
            run("UNTAR", alias:"untar_hto") {
                script "modules/nf-core/untar/main.nf"
                process {
                """
                input[0] = Channel.of([ [],file(params.modules_testdata_base_path + "/genomics/homo_sapiens/10xgenomics/cellranger/hashing_demultiplexing/hto.tar.gz", checkIfExists: true)   ])
                """
                }
            } 
            run("UNTAR", alias:"untar_rna") {
                script "modules/nf-core/untar/main.nf"
                process {
                """
                input[0] = Channel.of([ [],file(params.modules_testdata_base_path + "/genomics/homo_sapiens/10xgenomics/cellranger/hashing_demultiplexing/rna.tar.gz", checkIfExists: true)   ])
                """
                }
            }
        }
        when {
            process {
                """
                input[0] = UNTAR.out.untar_hto.map { meta, files -> [ [ id:'test', single_end:true ], files ] }
                input[1] = UNTAR.out.untar_rna.map { meta, files -> [ [ id:'test', single_end:true ], files ] }
                input[2] = false
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("Standard_Multiome - 10x mtx - stub") {
        config "./nextflow.config"
        options "-stub"

        when {
            process {
                """
                
                input[0] = [
                    [ id:'test'],
                    file(params.modules_testdata_base_path + "/genomics/homo_sapiens/10xgenomics/cellranger/hashing_demultiplexing/hto.tar.gz", checkIfExists: true),
                ]
                input[1] = [
                    [ id:'test'],
                    file(params.modules_testdata_base_path + "/genomics/homo_sapiens/10xgenomics/cellranger/hashing_demultiplexing/rna.tar.gz", checkIfExists: true),
                ]
                input[2] = false
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}

    

