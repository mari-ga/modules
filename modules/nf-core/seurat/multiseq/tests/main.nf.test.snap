{
    "Standard_Multiome - plots included - 10x mtx": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_assignment.csv:md5,66be2271ebac3a677d8b090e96b1029e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_classification.csv:md5,06390a70ca6e68aa78662dcb8719f708"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.R_sessionInfo.log:md5,a1b8e3a62e6470670c7afe1ca88b3234"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_ridge_plot.jpeg:md5,e69df314a77575d9c6566983b1d9e55f"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_feature_scatter_plot.jpeg:md5,590ca86fbdec0629cf216e79c16c023b"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_violin_plot.jpeg:md5,fee57b08580699622b95ad8a4b89e6ba"
                    ]
                ],
                "6": [
                    
                ],
                "7": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_tsne_classification.jpeg:md5,b421974f5d0522311e5bb8e61b4f544c"
                    ]
                ],
                "8": [
                    "versions.yml:md5,aea1fd7bd677abcbd5440531054e94e5"
                ],
                "assignment": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_assignment.csv:md5,66be2271ebac3a677d8b090e96b1029e"
                    ]
                ],
                "classification": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_classification.csv:md5,06390a70ca6e68aa78662dcb8719f708"
                    ]
                ],
                "featureScatter": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_feature_scatter_plot.jpeg:md5,590ca86fbdec0629cf216e79c16c023b"
                    ]
                ],
                "ridgePlot": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_ridge_plot.jpeg:md5,e69df314a77575d9c6566983b1d9e55f"
                    ]
                ],
                "session_info": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.R_sessionInfo.log:md5,a1b8e3a62e6470670c7afe1ca88b3234"
                    ]
                ],
                "tSNE_hto_Plot": [
                    
                ],
                "tsne_classification": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_tsne_classification.jpeg:md5,b421974f5d0522311e5bb8e61b4f544c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,aea1fd7bd677abcbd5440531054e94e5"
                ],
                "violingPlot": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_violin_plot.jpeg:md5,fee57b08580699622b95ad8a4b89e6ba"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-16T18:04:50.366498"
    }
}